lib/analytics/course_revisions_csv_builder.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CourseRevisionsCsvBuilder
+  def initialize(course)
+    @course = course
+    set_revisions
+  end
+
+  def generate_csv
+    csv_data = [CSV_HEADERS]
+    revisions_rows.each do |row|
+      csv_data << row
+    end
+    CSV.generate { |csv| csv_data.each { |line| csv << line } }
+  end
+
+  def revisions_rows
+    @new_revisions.values.map do |revision_data|
+      build_row(revision_data)
+    end
+  end
+
+  def set_revisions
+    @new_revisions = {}
+    @course.all_revisions.includes(article: :wiki).map do |edit|
+      revision_edits = @new_revisions[edit.article_id] || new_revision(edit)
+      revision_edits[:mw_rev_id] = edit.mw_rev_id
+      revision_edits[:mw_page_id] = edit.mw_page_id
+      revision_edits[:wiki_id] = edit.wiki_id
```

Comment:
Using `wiki_id` is a problem, as there's no way to see which wiki that ID is for. I think it needs to use `edit.wiki.domain` instead.